extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfRequestResponseReactiveFilter.java

```diff
@@ -96,18 +96,17 @@ public void filter(ResteasyReactiveContainerRequestContext requestContext, Routi
             }
         } else if (config.verifyToken) {
             // unsafe HTTP method, token is required
-
-            if (!isMatchingMediaType(requestContext.getMediaType(), MediaType.APPLICATION_FORM_URLENCODED_TYPE)
-                    && !isMatchingMediaType(requestContext.getMediaType(), MediaType.MULTIPART_FORM_DATA_TYPE)) {
+            MediaType mediaType = requestContext.getMediaType();
+            if (mediaType == null || (!MediaType.APPLICATION_FORM_URLENCODED_TYPE.isCompatible(mediaType)
```

Comment:
`isCompatible` is taking wilcards, which is why I individually compare types and subtypes,
```
public boolean isCompatible(final MediaType other) {
        ...
        return
            (type.equalsIgnoreCase(other.type) || this.isWildcardType() || other.isWildcardType())
            &&
            (subtype.equalsIgnoreCase(other.subtype) || this.isWildcardSubtype()
             || other.isWildcardSubtype());
    }
```
so if someone sends a `Content-Type: *` it will be marked as compatible which is a problem

Reply:
Ah thanks for the pointer. I've kept your method and added a Javadoc explaining why we do that instead